engine/src/flutter/shell/common/shell.cc

```diff
@@ -1072,15 +1072,59 @@ void Shell::OnPlatformViewScheduleFrame() {
                                     });
 }
 
+bool Shell::ValidateViewportMetrics(const ViewportMetrics& metrics) {
```

Comment:
Instead of a static method on `Shell`, I'd make this a top-level function in the anonymous namespace up above, after `PerformInitializationTasks`. This lets us keep this implementation detail out of the header file